roqoqo-qasm/src/backend.rs

```diff
@@ -366,11 +369,20 @@ impl Backend {
 #[derive(Debug, Clone, Copy, PartialEq, Eq)]
 pub enum QasmVersion {
     /// OpenQASM 2.0
-    V2point0,
+    V2point0(Qasm2Dialect),
     /// OpenQASM 3.0
     V3point0(Qasm3Dialect),
 }
 
+/// Enum for setting the version of OpenQASM used
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum Qasm2Dialect {
+    /// No Pragma operations
```

Comment:
I'd write simply `Vanilla OpenQASM 2.0`. The `pragma` keyword isn't even supported in 2.0, so saying that this is the dialect that doesn't contain it doesn't make too much sense.